tests/unit/test_session.py

```diff
@@ -2798,6 +3047,127 @@ def test_endpoint_from_production_variants_with_sagemaker_config_injection(
     )
 
 
+def test_endpoint_from_production_variants_with_sagemaker_config_injection_partial_kms_support(
```

Comment:
Thanks for adding these tests. Optional but I would prefer one or more comments for each test that make it clear what is being verified. Based on the title alone I can imagine it being difficult for someone reading through later to find whats different in the contents of one test vs another. 

Also, do we need to verify all the configs for these kms_support tests? I wonder if we could get rid of all the asserts that dont relate to the thing being tested?

Reply:
+1